### PR TITLE
[AIRFLOW-6025] Add label to uniquely identify creator of Pod

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6025

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
2 Components natively create Kubernetes Pods.

1) KubernetesPodOperator
2) KubernetesExecutor

It would be ideal to add a label that identifies which of the two launches pod.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by turbaszek** [comment]
It seems that we have a flaky test:
```
   Traceback (most recent call last):
    tests/task/task_runner/test_standard_task_runner.py line 129 in test_on_kill
      with open(path, "r") as f:
   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/airflow_on_kill'

   -------------------- >> begin captured stdout << ---------------------
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s
   [2019-11-21 13:02:57,945] {test_task_view_type_check.py:49} INFO - class_instance type: <class 'unusual_prefix_5d280a9b385120fec3c40cfe5be04e2f41b6b5e8_test_task_view_type_check.CallableClass'>
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s
   [%(asctime)s] {{%(filename)s:%(lineno)d}} %(levelname)s - %(message)s

   --------------------- >> end captured stdout << ----------------------
```

**Comment by kaxil** [comment]
@nuclearpinguin Yeah, restarting the test worked :)

File: airflow/contrib/operators/kubernetes_pod_operator.py [hunk 1]
```diff
@@ -127,6 +128,15 @@ def execute(self, context):
                                                  cluster_context=self.cluster_context,
                                                  config_file=self.config_file)
 
+            # Add Airflow Version to the label
+            # And a label to identify that pod is launched by KubernetesPodOperator
+            self.labels.update(
+                {
+                    'airflow_version': airflow_version.replace('+', '-'),
```
Comment (ashb) [inline airflow/contrib/operators/kubernetes_pod_operator.py:135]: Does Kube recommend that we qualify our labels, like `org.apache.airflow.version`?

Comment (ashb) [reply airflow/contrib/operators/kubernetes_pod_operator.py:135]: Or is that for annotations? When should we use one over another?

Comment (kaxil) [reply airflow/contrib/operators/kubernetes_pod_operator.py:135]: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ recommend some but given we already use labels and don't follow such conventions, I made this change :

https://github.com/apache/airflow/blob/fab957e763f40bf2a2398770312b4834fbd613e1/airflow/kubernetes/worker_configuration.py#L372-L378

Comment (kaxil) [reply airflow/contrib/operators/kubernetes_pod_operator.py:135]: Re. labels vs annotations:

>You can use either labels or annotations to attach metadata to Kubernetes objects. Labels can be used to select objects and to find collections of objects that satisfy certain conditions. In contrast, annotations are not used to identify and select objects. The metadata in an annotation can be small or large, structured or unstructured, and can include characters not permitted by labels.
So labels make more sense for e.g to select all the pods that are created by Kube Pod Operator

**Review by dimberman — APPROVED** [review]

**Review by ashb — APPROVED** [review]

**Pull request merged**